Modules/ThirdParty/Eigen3/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+project(ITKEigen3)
+set(ITKEigen3_THIRD_PARTY 1)
+set(ITKEigen3_Libraries ITKEigen3)
+set(ITKEigen3_NO_SRC 1)
+
+option(ITK_USE_SYSTEM_EIGEN "Use an outside build of Eigen." ${ITK_USE_SYSTEM_LIBRARIES})
+mark_as_advanced(ITK_USE_SYSTEM_EIGEN)
+
+
+if(ITK_USE_SYSTEM_EIGEN)
+  set(Eigen3_DIR_INSTALL ${Eigen3_DIR})
+  set(Eigen3_DIR_BUILD ${Eigen3_DIR})
+  set(Eigen3_DIR ${Eigen3_DIR_BUILD})
+  find_package(Eigen3 REQUIRED)
+else()
+  set(Eigen3_DIR_INSTALL "${CMAKE_INSTALL_PREFIX}/${ITK_INSTALL_LIBRARY_DIR}/cmake/ITK-${ITK_VERSION_MAJOR}.${ITK_VERSION_MINOR}/Modules")
+  set(Eigen3_DIR_BUILD "${CMAKE_CURRENT_BINARY_DIR}/src/itkeigen")
```

Comment:
itkeigen, not itkeigen**3**

Reply:
This was to minimize changes from the `for/vtk` fork. MR opened: https://gitlab.kitware.com/third-party/eigen/merge_requests/1

Reply:
All the work on VXL 2.0.0 was to tee up the possibility to removing VXL.

This is exciting.  Previously we have spoken about this transition is the primary aim of ITKv6.

Reply:
That's great @hjmjohnson! This could be a first step to just using Eigen3 to speed up some heavy lifting, like the eigen decomposition. Eigen3 has this Map concept that can work with pointers to arrays without a copy, so it is easy to use with VXL when the pointer to data is available, eg `data_block()`. Done in SymmetricEigenAnalysis in this PR.